File: src/test/java/social/alone/server/auth/FacebookUserInfoFetcherTest.java
package social.alone.server.auth;


import org.junit.Ignore;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;
import social.alone.server.auth.oauth2.user.FacebookOAuth2UserInfo;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest
@RunWith(SpringRunner.class)
public class FacebookUserInfoFetcherTest {

    @Autowired
    private FacebookUserInfoFetcher userInfoFetcher;

    @Test
    @Ignore
    public void test() {
        String accessToken = "foo";
        FacebookOAuth2UserInfo user = userInfoFetcher.getUserInfo(accessToken);
        assertThat(user.getEmail()).isNotEmpty();
        assertThat(user.getId()).isNotEmpty();
        assertThat(user.getName()).isNotEmpty();
        assertThat(user.getImageUrl()).isNotEmpty();
    }

}

File: src/test/kotlin/social/alone/server/auth/FacebookUserInfoFetcherTest.kt
package social.alone.server.auth


import org.junit.Ignore
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner
import social.alone.server.auth.oauth2.user.FacebookOAuth2UserInfo

import org.assertj.core.api.Assertions.assertThat

@SpringBootTest
@RunWith(SpringRunner::class)
class FacebookUserInfoFetcherTest {

    @Autowired
    lateinit var userInfoFetcher: FacebookUserInfoFetcher

    @Test
    @Ignore
    fun test() {
        val accessToken = "foo"
        val user = userInfoFetcher.getUserInfo(accessToken)
        assertThat(user.email).isNotEmpty()
        assertThat(user.id).isNotEmpty()
        assertThat(user.name).isNotEmpty()
        assertThat(user.imageUrl).isNotEmpty()
    }

}